WalletWasabi.Fluent/Controls/CurrencyEntryBox.axaml.cs

```diff
@@ -209,71 +210,24 @@ private void InsertLeadingZeroForDecimal(TextInputEventArgs e)
 		CaretIndex += prependText.Length;
 	}
 
-	[GeneratedRegex($"^(?<Whole>[0-9{GroupSeparator}]*)(\\{DecimalSeparator}?(?<Frac>[0-9{GroupSeparator}]*))$")]
-	private static partial Regex RegexBtcFormat();
-
-	[GeneratedRegex($"^[0-9{GroupSeparator}{DecimalSeparator}]*$")]
-	private static partial Regex RegexDecimalCharsOnly();
-
-	[GeneratedRegex($"{GroupSeparator}{{2,}}")]
-	private static partial Regex RegexConsecutiveSpaces();
-
-	[GeneratedRegex($"[{GroupSeparator}{DecimalSeparator}]+")]
-	private static partial Regex RegexGroupAndDecimal();
-
-	private bool ValidateEntryText(string preComposedText)
+	private void ReplaceCommandWithDot(TextInputEventArgs e)
 	{
-		// Check if it has a decimal separator.
-		var trailingDecimal = preComposedText.Length > 0 && preComposedText.EndsWith(DecimalSeparator);
-		var match = RegexBtcFormat().Match(preComposedText);
-
-		// Ignore group chars on count of the whole part of the decimal.
-		var wholeStr = match.Groups["Whole"].ToString();
-		var whole = RegexGroupAndDecimal().Replace(wholeStr, "").Length;
-
-		var fracStr = match.Groups["Frac"].ToString().Replace(GroupSeparator, "");
-		var frac = RegexGroupAndDecimal().Replace(fracStr, "").Length;
-
-		// Check for consecutive spaces (2 or more) and leading spaces.
-		var rule1 = preComposedText.Length > 1 && (preComposedText.StartsWith(GroupSeparator) ||
-												   RegexConsecutiveSpaces().IsMatch(preComposedText));
-
-		// Check for trailing spaces in the whole number part and in the last part of the precomp string.
-		var rule2 = whole >= 8 && (preComposedText.EndsWith(GroupSeparator) || wholeStr.EndsWith(GroupSeparator));
-
-		// Check for non-numeric chars.
-		var rule3 = !RegexDecimalCharsOnly().IsMatch(preComposedText);
-		if (rule1 || rule2 || rule3)
-		{
-			return false;
-		}
-
-		// Reject and dont process the input if the string doesnt match.
-		if (!match.Success)
-		{
-			return false;
-		}
-
-		// Passthrough the decimal place char or the group separator.
-		if (preComposedText == DecimalSeparator && !trailingDecimal)
-		{
-			return false;
-		}
+		var finalText = e.Text?.Replace(',', '.');
```

Comment:
https://github.com/zkSNACKs/WalletWasabi/blob/a45def074ba9c540859ae4e3bad596aad644af64/WalletWasabi/Userfacing/CurrencyInput.cs#L8

This method should be used for correction, because it deals with many other separators and formatting stuff.